### ISSUE TITLE
[BUG] Cannot discover GPU for execution

### ISSUE THREAD

**Issue by jiquant** [mars-project/mars#1370]
pymars 0.4.2

```
import mars.tensor as mt
a = mt.random.rand(10, 10, gpu=True)  
a.sum().execute()  
    
Traceback (most recent call last):

  File "<ipython-input-3-ead6efc7da0c>", line 1, in <module>
    a.sum().execute()

  File "D:\Anaconda3\lib\site-packages\mars\core.py", line 561, in execute
    self._data.execute(session, **kw)

  File "D:\Anaconda3\lib\site-packages\mars\core.py", line 370, in execute
    session.run(self, **kw)

  File "D:\Anaconda3\lib\site-packages\mars\session.py", line 428, in run
    result = self._sess.run(*tileables, **kw)

  File "D:\Anaconda3\lib\site-packages\mars\session.py", line 98, in run
    raise RuntimeError('No GPU found for execution')

RuntimeError: No GPU found for execution
```

**Comment by qinxuye** [comment]
For now, Mars just try to load `libnvidia-ml.so` to discover GPU, but sometimes the library path is not included in LD_LIBRARY_PATH. We need to try to optimize the seek path.

**Comment by ShiLijuan1997** [comment]
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda-10.0/targets/x86_64-linux/lib
i did it in this way, but still not work...

**Comment by qinxuye** [comment]
Could you please type

```
ls /usr/local/cuda-10.0/targets/x86_64-linux/lib
```

to see the result.

**Comment by ShiLijuan1997** [comment]
root@ctpai-node1:/# ls /usr/local/cuda-10.0/targets/x86_64-linux/lib
libOpenCL.so                  libcufftw_static.a       libnppial.so            libnppig_static.a      libnvToolsExt.so.1.0.0
libOpenCL.so.1                libcuinj64.so            libnppial.so.10.0       libnppim.so            libnvblas.so
libOpenCL.so.1.1              libcuinj64.so.10.0       libnppial.so.10.0.130   libnppim.so.10.0       libnvblas.so.10.0
libaccinj64.so                libcuinj64.so.10.0.130   libnppial_static.a      libnppim.so.10.0.130   libnvblas.so.10.0.130
libaccinj64.so.10.0           libculibos.a             libnppicc.so            libnppim_static.a      libnvgraph.so
libaccinj64.so.10.0.130       libcurand.so             libnppicc.so.10.0       libnppist.so           libnvgraph.so.10.0
libcublas.so                  libcurand.so.10.0        libnppicc.so.10.0.130   libnppist.so.10.0      libnvgraph.so.10.0.130
libcublas.so.10.0             libcurand.so.10.0.130    libnppicc_static.a      libnppist.so.10.0.130  libnvgraph_static.a
libcublas.so.10.0.130         libcurand_static.a       libnppicom.so           libnppist_static.a     libnvjpeg.so
libcublas_static.a            libcusolver.so           libnppicom.so.10.0      libnppisu.so           libnvjpeg.so.10.0
libcudadevrt.a                libcusolver.so.10.0      libnppicom.so.10.0.130  libnppisu.so.10.0      libnvjpeg.so.10.0.130
libcudart.so                  libcusolver.so.10.0.130  libnppicom_static.a     libnppisu.so.10.0.130  libnvjpeg.so.10.0.318
libcudart.so.10.0             libcusolver_static.a     libnppidei.so           libnppisu_static.a     libnvjpeg_static.a
libcudart.so.10.0.130         libcusparse.so           libnppidei.so.10.0      libnppitc.so           libnvrtc-builtins.so
libcudart_static.a            libcusparse.so.10.0      libnppidei.so.10.0.130  libnppitc.so.10.0      libnvrtc-builtins.so.10.0
libcufft.so                   libcusparse.so.10.0.130  libnppidei_static.a     libnppitc.so.10.0.130  libnvrtc-builtins.so.10.0.130
libcufft.so.10.0              libcusparse_static.a     libnppif.so             libnppitc_static.a     libnvrtc.so
libcufft.so.10.0.145          liblapack_static.a       libnppif.so.10.0        libnpps.so             libnvrtc.so.10.0
libcufft_static.a             libmetis_static.a        libnppif.so.10.0.130    libnpps.so.10.0        libnvrtc.so.10.0.130
libcufft_static_nocallback.a  libnppc.so               libnppif_static.a       libnpps.so.10.0.130    stubs
libcufftw.so                  libnppc.so.10.0          libnppig.so             libnpps_static.a
libcufftw.so.10.0             libnppc.so.10.0.130      libnppig.so.10.0        libnvToolsExt.so
libcufftw.so.10.0.145         libnppc_static.a         libnppig.so.10.0.130    libnvToolsExt.so.1

this is the result , and actually the libnvidia-ml.so is in the stubs directory, i also tried to export it, but still not work, and i found i also have a libnvidia-ml.so.1 in /usr/lib/x86_64-linux-gnu, i tried to export it, but not work.

**Comment by qinxuye** [comment]
Could you print the `LD_LIBRARY_PATH`? @tqslj2

**Comment by tqslj2** [comment]
root@ctpai-node1:/usr/local/cuda-10.0# echo $LD_LIBRARY_PATH
/usr/local/cuda-10.0/targets/x86_64-linux/lib:/usr/local/cuda-10.0/targets/x86_64-linux/lib:/usr/lib/x86_64-linux-gnu

this is all i have, tks.

**Comment by qinxuye** [comment]
@tqslj2 , Could you please run code below:

```bash
python -c "from ctypes import CDLL; CDLL('libnvidia-ml.so')"
```

and see what the output is.

**Comment by tqslj2** [comment]
root@ctpai-node1:~# python -c "from ctypes import CDLL; CDLL('libnvidia-ml.so')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/conda/envs/rapids/lib/python3.6/ctypes/__init__.py", line 348, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: libnvidia-ml.so: cannot open shared object file: No such file or directory

it shows this message, seems i didin't export the lib succesfully ?

**Comment by qinxuye** [comment]
@tqslj2 Yeah, it seems so. Looks like cannot find the .so file.

**Comment by qinxuye** [comment]
Check the link https://askubuntu.com/questions/833862/nvidia-smi-couldnt-find-libnvidia-ml-so-library to see if it can help or not.

**Comment by qinxuye** [comment]
This issue must have been resolved in v0.4.6 as well as v0.5.0rc1.

`pip install pymars>=0.4.6` to verify.
